In [1]:
import pandas as pd
import numpy as np



In [2]:
admission= pd.read_excel("D:\Datascience\Admission.xlsx")

In [3]:
admission.head()

admit  gre   gpa  rank
0      0  380  3.61     3
1      1  660  3.67     3
2      1  800  4.00     1
3      1  640  3.19     4
4      0  520  2.93     4

In [4]:
import seaborn as sns

In [5]:
admission.isnull().sum()

admit    0
gre      0
gpa      0
rank     0
dtype: int64

In [77]:
x=admission.drop(columns=['admit'])

In [78]:
x

gre   gpa  rank
0    380  3.61     3
1    660  3.67     3
2    800  4.00     1
3    640  3.19     4
4    520  2.93     4
..   ...   ...   ...
395  620  4.00     2
396  560  3.04     3
397  460  2.63     2
398  700  3.65     2
399  600  3.89     3

[400 rows x 3 columns]

In [79]:
y=admission[['admit']]

In [80]:
y

admit
0        0
1        1
2        1
3        1
4        0
..     ...
395      0
396      0
397      0
398      0
399      0

[400 rows x 1 columns]

In [81]:
from sklearn.model_selection import train_test_split

In [82]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.3,random_state=42)

In [83]:
len(x_train),len(x_test),len(y_train),len(y_test)

(280, 120, 280, 120)

In [84]:
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier

In [85]:
lr = LogisticRegression()
model_lr = lr.fit(x_train,y_train)

C:\Users\Mrunmayee\anaconda3\lib\site-packages\sklearn\utils\validation.py:72: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  return f(**kwargs)


In [86]:
y_test

admit
209      0
280      0
33       1
210      0
93       0
..     ...
60       1
79       1
285      0
305      0
281      0

[120 rows x 1 columns]

In [87]:
y_test['Predict_LR'] = model_lr.predict(x_test)

<ipython-input-87-d6b10bc4e691>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  y_test['Predict_LR'] = model_lr.predict(x_test)


In [88]:
y_test

admit  Predict_LR
209      0           0
280      0           0
33       1           0
210      0           0
93       0           0
..     ...         ...
60       1           0
79       1           1
285      0           0
305      0           0
281      0           0

[120 rows x 2 columns]

In [89]:
from sklearn.metrics import classification_report,confusion_matrix

In [90]:
print(classification_report(y_test['admit'],y_test['Predict_LR']))

              precision    recall  f1-score   support

           0       0.72      0.98      0.83        81
           1       0.80      0.21      0.33        39

    accuracy                           0.73       120
   macro avg       0.76      0.59      0.58       120
weighted avg       0.74      0.72      0.66       120



In [91]:
print(confusion_matrix(y_test['admit'],y_test['Predict_LR']))

[[79  2]
 [31  8]]


## The accuracy for logistic regression is 68% for the given model.

###  b)	For a one unit increase in GPA, the chance of being admitted to graduate school increases by what amount?                                                                                                                          

#### y=constant + m1x1 +m2x2+m3x3+...)  y is admit x1=gre x2=gpa x3=rank

In [96]:
print(model_lr.coef_)

[[ 0.00168434  0.76601343 -0.50764788]]


In [97]:
print(model_lr.intercept_)

[-3.17311734]


### Profit = -3.17311734 +(0.00168434 *gre)+(0.76601343*gpa)+(-0.50764788*rank)

### For a one unit increase in GPA, the chance of being admitted to graduate school increases by  0.76601343 amount

### c)	For a one unit increase in GRE, the chance of being admitted to graduate school increases by what amount? 

####  For a one unit increase in GRE, the chance of being admitted to graduate school increases by 0.00168434 amount

### e)	Using your model predict if a candidate with Rank = 3, GRE = 180 and GPA =2.1 will get the admission or not?                                                                                                                     

In [94]:
pred_data=pd.DataFrame({"rank":[3],
                       "gre":[180],
                       "gpa":[2.1]})
pred_data

rank  gre  gpa
0     3  180  2.1

In [95]:
pred_admit=model_lr.predict(pred_data)
pred_admit

array([1], dtype=int64)